Content.Server/Salvage/SalvageSystem.cs

```diff
@@ -0,0 +1,310 @@
+        }
+
+        private void OnInteractHand(EntityUid uid, SalvageMagnetComponent sm, InteractHandEvent args)
+        {
+            if (args.Handled)
+                return;
+            args.Handled = true;
+            _popupSystem.PopupEntity(CallSalvage(), uid, Filter.Entities(args.User.Uid));
+        }
+
+        private bool TryGetSalvagePlacementLocation(out MapCoordinates coords, out Angle angle)
+        {
+            coords = MapCoordinates.Nullspace;
+            angle = Angle.Zero;
+            foreach (var (smc, tsc) in EntityManager.EntityQuery<SalvageMagnetComponent, TransformComponent>(true))
+            {
+                coords = new EntityCoordinates(smc.OwnerUid, smc.Offset).ToMap(EntityManager);
+                var grid = tsc.GridID;
+                if (grid != GridId.Invalid)
+                {
+                    // Has a valid grid - synchronize angle so that salvage doesn't have to deal with cross-grid manipulation issues
+                    angle = _mapManager.GetGrid(grid).WorldRotation;
+                }
+                return true;
+            }
+            return false;
+        }
+
+        private IEnumerable<SalvageMapPrototype> GetAllSalvageMaps()
+        {
+            return _prototypeManager.EnumeratePrototypes<SalvageMapPrototype>();
+        }
+
+        // String is announced
+        private string SpawnSalvage()
+        {
+            // In case of failure
+            State = SalvageSystemState.Inactive;
+
+            if (!TryGetSalvagePlacementLocation(out var spl, out var spAngle))
+            {
+                return Loc.GetString("salvage-system-announcement-spawn-magnet-lost");
+            }
+
+            SalvageMapPrototype? map = null;
+
+            var forcedSalvage = _configurationManager.GetCVar<string>(CCVars.SalvageForced);
+            List<SalvageMapPrototype> allSalvageMaps;
+            if (string.IsNullOrWhiteSpace(forcedSalvage))
+            {
+                allSalvageMaps = GetAllSalvageMaps().ToList();
+            }
+            else
+            {
+                allSalvageMaps = new();
+                if (_prototypeManager.TryIndex<SalvageMapPrototype>(forcedSalvage, out map))
+                {
+                    allSalvageMaps.Add(map);
+                }
+                else
+                {
+                    Logger.ErrorS("c.s.salvage", $"Unable to get forced salvage map prototype {forcedSalvage}");
+                }
+            }
+            for (var i = 0; i < allSalvageMaps.Count; i++)
+            {
+                map = _random.PickAndTake(allSalvageMaps);
+
+                var box2 = Box2.CenteredAround(spl.Position, new Vector2(map.Size * 2.0f, map.Size * 2.0f));
+                var box2rot = new Box2Rotated(box2, spAngle, spl.Position);
+                if (_physicsSystem.GetCollidingEntities(spl.MapId, in box2rot).Select(x => EntityManager.HasComponent<IMapGridComponent>(x.OwnerUid)).Count() > 0)
```

Comment:
Didn't spot this last time but this is just a significantly slower "FindGridsIntersecting"